lib/amazonka-core/src/Amazonka/Sign/V4.hs

```diff
@@ -42,7 +43,16 @@ presign ex rq a r ts = Base.signRequest meta mempty auth
         . pair (CI.original hAMZSignedHeaders) (toBS shs)
         . pair (CI.original hAMZToken) (toBS <$> _authSessionToken a)
 
-    digest = Base.Tag "UNSIGNED-PAYLOAD"
+    digest =
+      case _requestBody rq of
+        Chunked _ -> unsignedPayload
+        Hashed HashedStream {} -> unsignedPayload
+        Hashed (HashedBytes _ b) ->
+          if BS.length b == 0 && (_serviceSigningName . _requestService $ rq) == "s3"
```

Comment:
Use `BS.null b`, and I think this would read nicer if you used guards.

Reply:
Done